File: BungiiAutomation/Bungii.Test.Regression.Web.Integration/Pages/Driver/Driver_ForgotPasswordPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace Bungii.Android.Regression.Test.Integration.Pages.Driver
{
    public class Driver_ForgotPasswordPage
    {
        public Driver_ForgotPasswordPage(IWebDriver webdriver)
        {
            PageFactory.InitElements(webdriver, this);
        }

        //---------------------------Forgot Password--------------------------------------------------------------------
        //Driver Login - Forgot Password Link
        [FindsBy(How = How.XPath, Using = "//form[@id='Login']/div/a[text()='Forgot Password?']")]
        public IWebElement Link_ForgotPassword { get; set; }

        //Driver Forgot Password - Header
        [FindsBy(How = How.XPath, Using = "//div[@id='forgot-pwd']/h2")]
        public IWebElement Header_ForgotPassword { get; set; }

        //Driver Forgot Password - Blank field validation error
        [FindsBy(How = How.Id, Using = "forgotpassworderrorsummary")]
        public IWebElement Err_ForgotPass_BlankField { get; set; }

        //Driver Forgot Password - Phone Number field
        [FindsBy(How = How.XPath, Using = "//form[@id='ForgotPassword']/div/input[@id='PhoneNo']")]
        public IWebElement Textfield_ForgotPass_Phone { get; set; }

        //Driver Forgot Password - Phone Number field Error
        [FindsBy(How = How.Id, Using = "PhoneNo-error")]
        public IWebElement Err_ForgotPass_Phone { get; set; }

        //Driver Forgot Password - Send Verification Code button
        [FindsBy(How = How.XPath, Using = "//form[@id='ForgotPassword']/button[contains(text(),'Code')]")]
        public IWebElement Button_SendVerifCode { get; set; }

        //Driver Forgot Password - Back to Login link
        [FindsBy(How = How.XPath, Using = "//form[@id='ForgotPassword']/div/a[text()='Back to Login']")]
        public IWebElement Link_BackToLogin { get; set; }
    }
}

File: Bungii.Test.Integration/src/test/java/com/bungii/web/pages/driver/Driver_ForgotPasswordPage.java
package com.bungii.web.pages.driver;

import com.bungii.common.core.PageBase;
import org.openqa.selenium.WebElement;

public class Driver_ForgotPasswordPage extends PageBase {

    //---------------------------Forgot Password--------------------------------------------------------------------
    //driver Login - Forgot Password Link
    public WebElement Link_ForgotPassword () { return findElement("//a[contains(text(),'forgot password?')]", LocatorType.XPath); }

    //driver Forgot Password - Header
    public WebElement Header_ForgotPassword () { return findElement("//body/div/div/div/div/div/aside/div/div[3]/form/h2", LocatorType.XPath); }

    //driver Forgot Password - Blank field validation error
    public WebElement Err_ForgotPass_BlankField () { return findElement("//p[contains(text(),'Oops! It looks like you missed something. Please f')]", LocatorType.XPath); }

    //driver Forgot Password - Phone Number field
    public WebElement Textfield_ForgotPass_Phone() { return findElement("//div[@id='root']/div/div/div/div/aside/div/div[3]/form/div/input", LocatorType.XPath); }

    //driver Forgot Password - Phone Number field Error
    public WebElement Err_ForgotPass_Phone() { return findElement("//div[@id='root']/div/div/div/div/aside/div/div[3]/form/div/div", LocatorType.XPath); }

    //driver Forgot Password - Send Verification Code button
    public WebElement Button_SendVerifCode() { return findElement("//button[contains(text(),'Send Verification Code')]", LocatorType.XPath); }

    //driver Forgot Password - Back to Login link
    public WebElement Link_BackToLogin() { return findElement("uncontrolled-tab-example-tab-login", LocatorType.Id); }
}
